From: Simon McVittie <smcv@collabora.com>
Date: Tue, 13 Feb 2024 10:19:36 +0000
Subject: gdatetime: Exercise %c, %C format placeholders in ja_JP.eucjp

Previously we didn't test these at all, which made it hard to determine
whether %Ec, %EC had appropriate output on platforms where era-based
dates are unsupported. Now we do, and we can observe that on platforms
with no support for era-based dates, %c matches %Ec and %C matches %EC,
as desired.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Forwarded: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3920
---
 glib/tests/gdatetime.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/glib/tests/gdatetime.c b/glib/tests/gdatetime.c
index f2de1ab..00fb963 100644
--- a/glib/tests/gdatetime.c
+++ b/glib/tests/gdatetime.c
@@ -1682,6 +1682,8 @@ test_non_utf8_printf (void)
   TEST_PRINTF ("%b", "10\346\234\210");
 #endif
   TEST_PRINTF ("%B", "10\346\234\210");
+  TEST_PRINTF ("%c", "2009年10月24日 00時00分00秒");
+  TEST_PRINTF ("%C", "20");
   TEST_PRINTF ("%d", "24");
   TEST_PRINTF_DATE (2009, 1, 1, "%d", "01");
   TEST_PRINTF ("%e", "24"); // fixme
